### ISSUE TITLE
Incorrect error message with malformed Fn::Invoke

### ISSUE THREAD

**Issue by joeduffy** [pulumi/pulumi-yaml#254]
Oddly, if I have the following:

```yaml
name: pulumi-ansible-wordpress
runtime: yaml
description: Deploy an EC2 Wordpress server using Pulumi and Ansible.

configuration:
  publicKeyPath:
    type: String
    required: true
  privateKeyPath:
    type: String
    required: true
  dbInstanceSize:
    type: String
    default: db.t3.small
  dbName:
    type: String
    default: wordpressdb
  dbUsername:
    type: String
    default: admin
  dbPassword:
    type: String
    required: true
  ec2InstanceSize:
    type: String
    default: t3.small

variables:
  # Dynamically fetch AZs so we can spread across them.
  availabilityZones:
    Fn::Invoke:
      Function: aws:getAvailabilityZones
  # Dynamically query for the Amazon Linux AMI in this region.
  awsLinuxAmi:
    Fn::Invoke:
      function: aws:ec2:getAmi
      arguments:
        filters:
          - name: name
            values: ["amzn-ami-hvm-*-x86_64-ebs"]
        owners: ["137112412989"] # Amazon
        mostRecent: true

...
```

I get the following error: `error: failed to discover plugin requirements: Pulumi.yaml:8,5-13: Field 'required' does not exist on Object 'configuration.publicKeyPath'; Existing fields are: 'secret', 'type', 'default', and 3 other diagnostic(s)`.

The thing is this seems to have nothing to do with the real problem. The problem is that the second `Fn::Invoke` uses lowercase `function` and `argument` properties. They need to be uppercase, as in `Function` and `Argument`.

Indeed, fixing that makes the seemingly unrelated error go away!

**Comment by mikhailshilkov** [comment]
As mentioned in https://github.com/pulumi/pulumi/issues/9890, `required` is not a valid property, so the error is legit. You can also see `and 3 other diagnostic(s)` which hides your actual error. Once I fix the required warnings, I get `could not load yaml program: failed to load the template: Pulumi.yaml:35,7-41,25: missing function name ('Function'); missing function name ('Function')`.

I suppose we can use this issue to track improvements to error presentations. Why do we need to hide the warnings under "and 3 other diagnostic(s)"? Can we sort the diagnostics so that errors are ahead of warnings?

**Comment by AaronFriel** [comment]
I think after landing the fix to #253 in PR #258, we should show all diagnostics.
